examples/02enum/main.go

```diff
@@ -0,0 +1,76 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"reflect"
+	"strings"
+
+	"github.com/podhmo/structflag"
+)
+
+type Options struct {
+	Name      string    `flag:"name" help:"name of greeting"`
+	LogLevel  LogLevel  `flag:"log-level"`
+	LogLevel2 *LogLevel `flag:"log-level2"`
+}
+
+func main() {
+	defaultLogLevel := LogLevelInfo
+	options := &Options{Name: "foo", LogLevel: defaultLogLevel, LogLevel2: &defaultLogLevel} // default value
+
+	b := structflag.NewBuilder()
+	b.Name = "hello"
+	b.EnvPrefix = "X_"
+
+	fs := b.Build(options)
+	fs.Parse(os.Args[1:])
+
+	fmt.Printf("parsed: %#+v\n", options)
+
+}
+
+type LogLevel string
+
+const (
+	LogLevelDebug   LogLevel = "DEBUG"
+	LogLevelInfo    LogLevel = "INFO"
+	LogLevelWarning LogLevel = "WARN"
+	LogLevelError   LogLevel = "ERROR"
+)
+
+func (v LogLevel) Validate() error {
+	switch v {
+	case "DEBUG", "INFO", "WARN", "ERROR":
+		return nil
+	default:
+		return fmt.Errorf("%v is an invalid value for %v", v, reflect.TypeOf(v))
+	}
+}
+
+// for structflag.HasHelpText
+func (v LogLevel) HelpText() string {
+	return "log level {DEBUG, INFO, WARN, ERROR}"
+}
+
+// for pflag.Value
```

Comment:
for pflag.Value, the receiver of method should be pointer. (for also support pointer value)